go/pkg/rexec/rexec.go

```diff
@@ -0,0 +1,214 @@
+// Package rexec provides a top-level client for executing remote commands.
+package rexec
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/bazelbuild/remote-apis-sdks/go/digest"
+	"github.com/bazelbuild/remote-apis-sdks/go/pkg/command"
+	"github.com/bazelbuild/remote-apis-sdks/go/pkg/outerr"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	rc "github.com/bazelbuild/remote-apis-sdks/go/client"
+	repb "github.com/bazelbuild/remote-apis/build/bazel/remote/execution/v2"
+	log "github.com/golang/glog"
+)
+
+// FileDigestCache is a cache for file contents->digests.
+type FileDigestCache interface {
+	Get(string) (digest.Digest, error)
+}
+
+// NoopFileDigestCache is a non-caching cache (always returns a cache miss).
+type NoopFileDigestCache struct{}
+
+// Get computes the digest from the file contents.
+func (c *NoopFileDigestCache) Get(path string) (digest.Digest, error) {
+	return digest.NewFromFile(path)
+}
+
+// Client is a remote execution client.
+type Client struct {
+	FileDigestCache FileDigestCache
```

Comment:
Do you use the FileDigestCache anywhere?

Reply:
Alas, not yet. It's coming, but at least 4 patches away.